Project/DAQ_System/Program/Src/Sensor/Accelerometer/LSM6DSOXAccelerometer.cpp

```diff
@@ -0,0 +1,206 @@
+/*
+* Accelerometer LSM6DSOX
+* Authors:       William Lim
+*
+* Email:         wll180000@utdallas.edu
+*
+* (c) 2023 Dallas Formula Racing - Embedded Firmware Team
+* Formula SAE International Collegiate Chapter
+* GPL-3.0 License
+*/
+// Standard Libraries
+#include <stdint.h>
+#include <cmath>
+#include <limits>
+
+// DFR Custom dependencies
+#include "LSM6DSOXAccelerometer.hpp"
+
+
+
+namespace sensor{
+
+LSM6DSOX::LSM6DSOX(I2C_HandleTypeDef& hi2c):i2c_(hi2c){}
+
+void LSM6DSOX::init() {
+    static constexpr uint8_t kNumBytes = 2;
+    uint8_t commands[kNumBytes] = {0};
+
+    // data rate 104hz
+    commands[0] = CTRL1_XL;
+    commands[1] = 0x40; // 0b01000000
+    HAL_I2C_Master_Transmit(&i2c_,ACC_ADDRESS, commands,kNumBytes, HAL_MAX_DELAY);
+
+    calibrate();
+}
+
+void LSM6DSOX::calibrate() {
+    static constexpr uint8_t kSampleCount = 100;
+    double acc_magnitude = 0;
+    double average_array[kSampleCount];
+
+    // Collects samples to calibrate for 1G
+    for(int i = 0; i < kSampleCount; i++) {
+        ReadRawAcceleration();
+        acc_magnitude = sqrt(raw_acceleration_data_[0] * raw_acceleration_data_[0] + raw_acceleration_data_[1] * raw_acceleration_data_[1] + raw_acceleration_data_[2] * raw_acceleration_data_[2]);
+        average_array[i] = acc_magnitude * 0.001;
+    }
+
+    double min_value = std::numeric_limits<double>::max();
+    double max_value = -1;
+
+    // Find the largest and smallest magnitudes
+    for(int j = 2; j < kSampleCount; j++) {
+        if(average_array[j] <= min_value) {
+            min_value = average_array[j];
+
+        } else if (average_array[j] > max_value) {
+            max_value = average_array[j];
+        }
+    }
```

Comment:
There is no need to store each value and go back over the entire array. 
```suggestion
    static constexpr uint8_t kSampleCount = 100;
    double acc_magnitude = 0;
    double min_value = std::numeric_limits<double>::max();
    double max_value = -1;

    // Collects samples to calibrate for 1G
    for(int i = 0; i < kSampleCount; i++) {
        ReadRawAcceleration();
        acc_magnitude = sqrt(raw_acceleration_data_[0] * raw_acceleration_data_[0] + raw_acceleration_data_[1] * raw_acceleration_data_[1] + raw_acceleration_data_[2] * raw_acceleration_data_[2]);
        acc_magnitude *= 0.001;
        
         if(acc_magnitude <= min_value) {
            min_value = acc_magnitude;

        } else if (acc_magnitude > max_value) {
            max_value = acc_magnitude;
        }
    }
```

Reply:
this portion of the code is not my code; honestly, I don't know how this code correlates to the Accelerometer library as a whole. also, I  don't use the gravity_adjustment_conversion_factor variable anywhere. So I'm thinking about removing this calibrate function. It could also be that I'm just not seeing the bigger picture for this calibrate function. If that is the case, I can probably edit it and make it less in efficient.